t4.py

```diff
@@ -1,125 +1,121 @@
 import requests
 import json
 import os
-
-# 1. Hardcoded sensitive credentials and secrets (BAD PRACTICE)
-DATABASE_USERNAME = "admin"  # Hardcoded username (BAD)
-DATABASE_PASSWORD = "password123"  # Hardcoded password (BAD)
-API_KEY = "my_secret_api_key_123"  # Hardcoded API key (BAD PRACTICE)
-SECRET_KEY = "my_secret_key_987"  # Hardcoded secret key (BAD)
+DATABASE_USERNAME = "admin"  
+DATABASE_PASSWORD = "password123"  
+API_KEY = "my_secret_api_key_123"  
+SECRET_KEY = "my_secret_key_987"  
 PRIVATE_KEY = """
 -----BEGIN PRIVATE KEY-----
 MIIBVgIBADANBgkqhkiG9w0BAQEFAASCATwwggE4AgEAAkEAhackable_private_key_content
 badsecretkeyexposedtoeveryone+++++++++++++++/+++++++more_bad_exposure++
 -----END PRIVATE KEY-----
-"""  # Exposed private key (BIG SECURITY RISK)
+"""  
 
-# 2. Weak encryption (XOR-based encryption) - BAD PRACTICE
 def weak_encrypt(data):
-    key = "weak_key_123"  # Weak encryption key (easy to guess)
+    key = "weak_key_123"  
     encrypted_data = ''.join(chr(ord(c) ^ ord(key[i % len(key)])) for i, c in enumerate(data))
     return encrypted_data
 
-# 3. Poor error handling with a broad exception catch
+
 def fetch_user_data(username):
-    print(f"Connecting to the database with username: {DATABASE_USERNAME} and password: {DATABASE_PASSWORD}")  # Exposing sensitive information in logs
+    print(f"Connecting to the database with username: {DATABASE_USERNAME} and password: {DATABASE_PASSWORD}")  
     try:
-        # Simulating insecure database query with hardcoded credentials
+        
         if username == "admin":
-            return {"username": "admin", "password": "plaintextpassword", "role": "admin"}  # Exposing plaintext passwords
+            return {"username": "admin", "password": "plaintextpassword", "role": "admin"}  
         elif username == "user":
-            return {"username": "user", "password": "userpassword", "role": "user"}  # No password protection (BAD)
+            return {"username": "user", "password": "userpassword", "role": "user"}  
         else:
-            # Return None for non-existent users, with no proper error message
+            
             print("User not found.")
     except Exception as e:
-        # Broad exception catch (BAD PRACTICE)
+        
         print("An error occurred:", e)
 
```

Comment:
Logging sensitive information such as usernames, passwords, and API keys can lead to security vulnerabilities by exposing this information in log files, which might not be securely stored. Always sanitize logs to ensure they do not contain sensitive information.

**Label**: security

**Code suggestion**

```python
# Sanitize logs to avoid exposing sensitive information
print(f"Attempting login for user: {username}")
print("Making an API request")
```